plugins/repository-gcs/src/main/java/org/opensearch/repositories/gcs/GoogleCloudStorageClientSettings.java

```diff
@@ -114,6 +120,54 @@
         key -> new Setting<>(key, "repository-gcs", Function.identity(), Setting.Property.NodeScope, Setting.Property.Deprecated)
     );
 
+    /** Proxy type */
+    static final Setting.AffixSetting<Proxy.Type> PROXY_TYPE_SETTING = Setting.affixKeySetting(
+        PREFIX,
+        "proxy.type",
+        (key) -> new Setting<Proxy.Type>(
+            key,
+            Proxy.Type.DIRECT.name(),
+            s -> Proxy.Type.valueOf(s.toUpperCase(Locale.ROOT)),
+            Setting.Property.NodeScope
+        )
+    );
+
+    /** The host of a proxy to connect */
+    static final Setting.AffixSetting<String> PROXY_HOST_SETTING = Setting.affixKeySetting(
+        PREFIX,
+        "proxy.host",
+        key -> Setting.simpleString(key, Setting.Property.NodeScope),
+        () -> PROXY_TYPE_SETTING
+    );
+
+    /** The port of a proxy to connect */
+    static final Setting.AffixSetting<Integer> PROXY_PORT_SETTING = Setting.affixKeySetting(
+        PREFIX,
+        "proxy.port",
+        key -> Setting.intSetting(key, 0, 0, 1 << 16, Setting.Property.NodeScope),
```

Comment:
Super minor - the `maxValue` condition should really be `(1 << 16) - 1`

Reply:
good point, will fix.